test/strings/basic.jl

```diff
@@ -428,6 +428,9 @@ foobaz(ch) = reinterpret(Char, typemax(UInt32))
 # issue #17624, missing getindex method for String
 @test "abc"[:] == "abc"
 
-# PR #18259 reverted because failing test below
-a = [x for x in String([0xcf, 0x83, 0x83, 0x83, 0x83])]
-@test a[1] == 'σ'
+# we assume for performance that next/nextind return values past the end of a
+# string's underlying data; this helps with performance of `done`.
```

Comment:
It won't throw an exception. It will just return `false` if there is still stuff to consume, even if that stuff cannot be read as a valid character. This is in contrast to the old behaviour, where `done` will return `true` if all remaining string data is junk.